bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java

```diff
@@ -132,7 +132,8 @@ private static Type getAvailableImmutableType(Type type) {
 
 
     private static Type getImmutableType(Type type, Set<Type> unresolvedTypes) {
-        if (TypeChecker.isInherentlyImmutableType(type)) {
+        if (TypeChecker.isInherentlyImmutableType(type) || (SymbolFlags.isFlagOn(type.getFlags(),
```

Comment:
Wondering if https://github.com/ballerina-platform/ballerina-lang/blob/80e4401bd9c0ce1f0059eee3111954d74d6d78b2/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java#L116 also needs to be updated. I'm not sure, may have to check that path.

Reply:
Shall we create an issue? Error unions are also effectively immutable. We need to check and sync the behaviour with compile-time behaviour if there are any differences, IMO.

Reply:
Added https://github.com/ballerina-platform/ballerina-lang/issues/32083